nrfx/hal/nrf_comp.h

```diff
@@ -481,12 +481,14 @@ NRF_STATIC_INLINE void nrf_comp_th_set(NRF_COMP_Type * p_reg, nrf_comp_th_t thre
 NRF_STATIC_INLINE void nrf_comp_main_mode_set(NRF_COMP_Type *      p_reg,
                                               nrf_comp_main_mode_t main_mode)
 {
+    p_reg->MODE &= ~(COMP_MODE_MAIN_Msk);
```

Comment:
I recommend to make it similar to for example spim:
https://github.com/zephyrproject-rtos/hal_nordic/blob/master/nrfx/hal/nrf_spim.h#L1524
or
https://github.com/zephyrproject-rtos/hal_nordic/blob/master/nrfx/hal/nrf_glitchdet.h#L103
`
p_reg->MODE = (p_reg->MODE & ~(COMP_MODE_MAIN_Msk)) | (main_mode << COMP_MODE_MAIN_Pos);
`

Reply:
done :)